provider.go

```diff
@@ -42,12 +48,51 @@ type ClientProvider struct {
 //     gate := g8.NewGate(g8.NewAuthorizationService().WithClientProvider(clientProvider))
 func NewClientProvider(getClientByTokenFunc func(token string) *Client) *ClientProvider {
 	return &ClientProvider{
-		cache:                false,
 		getClientByTokenFunc: getClientByTokenFunc,
 	}
 }
 
+// WithCache adds cache options to the ClientProvider
+// ttl is the time until the cache entry will be deleted. A ttl of -1 means no expiration
+// maxSize is the maximum amount of entries that can be in the cache at any given time. If a value of 0 or less is provided, it means
+// infinite
+//
+// Example:
+// 		 clientProvider := g8.NewClientProvider(func(token string) *g8.Client {
+//         // We'll assume that the following function calls your database and returns a struct "User" that
+//         // has the user's token as well as the permissions granted to said user
+//         user := database.GetUserByToken(token)
+//         if user != nil {
+//             return g8.NewClient(user.Token).WithPermissions(user.Permissions)
+//         }
+//         return nil
+// 			})
+// 			clientProvider.WithCache(60*time.Minute, 70000)
```

Comment:
I'm not sure for that example line 70. Wouldn't it be better to have:
```go
//   clientProvider := g8.NewClientProvider(func(token string) *g8.Client {
//         // We'll assume that the following function calls your database and returns a struct "User" that
//         // has the user's token as well as the permissions granted to said user
//         user := database.GetUserByToken(token)
//         if user != nil {
//             return g8.NewClient(user.Token).WithPermissions(user.Permissions)
//         }
//         return nil
//   }).WithCache(60*time.Minute, 70000)
//     gate := g8.NewGate(g8.NewAuthorizationService().WithClientProvider(clientProvider))

```

Reply:
@ynden Great idea! I think the latter is better.